api/v1beta2/sts_types.go

```diff
@@ -0,0 +1,23 @@
+/*
+Copyright 2024 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+const (
+	// AmazonSTSProvider represents the AWS provider for Security Token Service.
+	// Provides support for fetching temporary credentials from an AWS STS endpoint.
+	AmazonSTSProvider string = "aws"
```

Comment:
I created this constant decoupled from the Bucket API since we may want to use this for other AWS integrations in the future, e.g. for OCIRepository.

Reply:
> But one can argue that all the similar constants in this package have the provider name at the beginning and the common part at the end and we should continue with that pattern.

This was my initial reasoning, but I prefer starting with the common part and putting the variations at the end much more :+1: 

I can change to `STSProviderAmazon` in this PR and open an issue for renaming all symbols in v1.

Reply:
Fixed, thanks! Here's the issue for renaming on v1: https://github.com/fluxcd/source-controller/issues/1569